### PR TITLE
fix json encode (forcing UTF-8)

### LINKED ISSUE #164: UnicodeEncodeError: 'latin-1' codec can't encode characters in position 106-107: ordinal not in range(256)
ERROR:Sync Settings.sync_settings.libs.logger:'latin-1' codec can't encode characters in position 106-107: ordinal not in range(256)
Traceback (most recent call last):
  File "D:\软件\sublime Text\Data\Installed Packages\Sync Settings.sublime-package\sync_settings/commands/create_and_upload.py", line 45, in create
    ).create(data)
  File "D:\软件\sublime Text\Data\Installed Packages\Sync Settings.sublime-package\sync_settings/libs/gist.py", line 36, in auth_wrapper
    return func(self, *args, **kwargs)
  File "D:\软件\sublime Text\Data\Installed Packages\Sync Settings.sublime-package\sync_settings/libs/gist.py", line 66, in create
    return self.__do_request('post', self.make_uri(), data=sublime.encode_value(data, True)).json()
  File "D:\软件\sublime Text\Data\Installed Packages\Sync Settings.sublime-package\sync_settings/libs/gist.py", line 92, in __do_request
    response = getattr(requests, verb)(url, headers=self.headers, proxies=self.proxies, **kwargs)
  File "D:\软件\SUBLIM~2\Data\Packages\requests\all\requests\api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "D:\软件\SUBLIM~2\Data\Packages\requests\all\requests\api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "D:\软件\SUBLIM~2\Data\Packages\requests\all\requests\sessions.py", line 522, in request
    resp = self.send(prep, **send_kwargs)
  File "D:\软件\SUBLIM~2\Data\Packages\requests\all\requests\sessions.py", line 642, in send
    r = adapter.send(request, **kwargs)
  File "D:\软件\SUBLIM~2\Data\Packages\requests\all\requests\adapters.py", line 439, in send
    timeout=timeout
  File "D:\软件\SUBLIM~2\Data\Packages\requests\all\requests\packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "D:\软件\SUBLIM~2\Data\Packages\requests\all\requests\packages\urllib3\connectionpool.py", line 356, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "./python3.3/http/client.py", line 1065, in request
  File "./python3.3/http/client.py", line 1102, in _send_request
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 106-107: ordinal not in range(256)

### LINKED ISSUE #165: Failed to do a "Create and Upload" - in Ubuntu 20.04 Sublime Text
WARNING:Sync Settings.sync_settings.libs.logger:{'documentation_url': 'https://developer.github.com/v3/gists/#create-a-gist', 'message': 'Problems parsing JSON'}
ERROR:Sync Settings.sync_settings.libs.logger:Unexpected Error, Reason: Problems parsing JSON
Traceback (most recent call last):
  File "/home/gerard/.config/sublime-text-3/Installed Packages/Sync Settings.sublime-package/sync_settings/commands/create_and_upload.py", line 45, in create
    ).create(data)
  File "/home/gerard/.config/sublime-text-3/Installed Packages/Sync Settings.sublime-package/sync_settings/libs/gist.py", line 36, in auth_wrapper
    return func(self, *args, **kwargs)
  File "/home/gerard/.config/sublime-text-3/Installed Packages/Sync Settings.sublime-package/sync_settings/libs/gist.py", line 66, in create
    return self.__do_request('post', self.make_uri(), data=sublime.encode_value(data, True)).json()
  File "/home/gerard/.config/sublime-text-3/Installed Packages/Sync Settings.sublime-package/sync_settings/libs/gist.py", line 104, in __do_request
    raise UnexpectedError('Unexpected Error, Reason: {}'.format(response.json()['message']))
Sync Settings.sync_settings.libs.gist.UnexpectedError: Unexpected Error, Reason: Problems parsing JSON

### PR DESCRIPTION
**Pull request by TheSecEng**
fixes #165 
fixes #164 

### Problem
gist requires valid JSON. JSON must be UTF-8 encoded in order to be valid.

### Solution

We encode out data to UTF-8 prior to upload.

### CODE REVIEW

**Comment by TheSecEng** [comment]
@mfuentesg this solves those problems. Or if you want to handle it exactly like `json` I can update this to escape unicode with a `\` which is what python json does.

File: sync_settings/libs/gist.py [hunk 1]
```diff
@@ -63,14 +65,14 @@ def make_uri(endpoint=''):
...
-        return self.__do_request('post', self.make_uri(), data=sublime.encode_value(data, True)).json()
+        return self.__do_request('post', self.make_uri(), data=sublime.encode_value(data, True).encode("utf-8")).json()
 
     @auth
     @with_gid
     def update(self, gid, data):
         if not isinstance(data, dict) or not len(data):
             raise ValueError('Gist can`t be updated without data')
-        return self.__do_request('patch', self.make_uri(gid), data=sublime.encode_value(data, True)).json()
+        return self.__do_request('patch', self.make_uri(gid), data=sublime.encode_value(data, True).encode("utf-8")).json()
```
Comment (mfuentesg) [inline sync_settings/libs/gist.py:75]: can you add a test replicating this escenario?

Comment (TheSecEng) [reply sync_settings/libs/gist.py:75]: @mfuentesg what kind of test do you want? Just ensure that non utf-8 characters are actually encoded?

Comment (mfuentesg) [reply sync_settings/libs/gist.py:75]: Usually i add tests to ensure that i am fixing the found issue.
In this case, we need to use an "invalid" file content, and the plugin must be able to encoding it.

Comment (TheSecEng) [reply sync_settings/libs/gist.py:75]: Gotcha, I will do that now

**Comment by TheSecEng** [comment]
Failed due to wrong order of dump but contain is the same. Hopefully this is how you meant.

**Comment by mfuentesg** [comment]
Hey @TheSecEng thanks for your proposal, but i am avoiding as max i can `sublime` or `sublime_api` functions, because are a black box to me, i don't know what is happening behind.

I will close this PR in favor or https://github.com/mfuentesg/SyncSettings/pull/168

**Comment by TheSecEng** [comment]
Sure, as long as we can handle comments I am fine with moving back to it. I will update my location pr. Sorry for any trouble with moving to sublime functions

**Comment by mfuentesg** [comment]
@TheSecEng  no problem, you are welcome to help me solving the open issues, because i don't have bandwidth at all hahaha.

**Pull request closed without merging**